### PR TITLE
Update UML diagram links in DG

### PR DESCRIPTION
**Pull request by OhJunMing**

### CODE REVIEW

**Comment by OhJunMing** [comment]
Still contains references to addressbook, because we have yet to rename addressbook

But this renaming of the class is not urgent, given that we have other tasks to complete

File: docs/diagrams/StorageClassDiagram.puml [hunk 1]
```diff
@@ -18,7 +18,7 @@ StorageManager o--> AddressBookStorage
 
 JsonUserPrefsStorage .left.|> UserPrefsStorage
 JsonAddressBookStorage .left.|> AddressBookStorage
-JsonAddressBookStorage .down.> JsonSerializableAddressBookStorage
-JsonSerializableAddressBookStorage .right.> JsonSerializablePerson
-JsonSerializablePerson .right.> JsonAdaptedTag
+JsonAddressBookStorage .down.> JsonSerializableAddressBook
+JsonSerializableAddressBook .right.> JsonAdaptedCustomer
+JsonAdaptedCustomer .right.> JsonAdaptedTag
```
Comment (OhJunMing) [inline docs/diagrams/StorageClassDiagram.puml:23]: https://github.com/nus-cs2103-AY2021S2/forum/issues/273

As per must-see issues in tp forum

Incorrect class name in storage class diagram

File: docs/diagrams/LogicClassDiagram.puml [hunk 2]
```diff
@@ -55,6 +61,10 @@ LogicManager --> Model
 Command .right.> Model
 note right of XYZCommand: XYZCommand = AddCommand, \nFindCommand, etc
 
+LogicManager --> Storage
+Storage --[hidden] Model
+Command .[hidden]up.> Storage
```
Comment (OhJunMing) [inline docs/diagrams/LogicClassDiagram.puml:66]: https://github.com/nus-cs2103-AY2021S2/forum/issues/257

Logic Manager should be aware of Storage

File: docs/diagrams/StorageClassDiagram.puml [hunk 3]
```diff
@@ -16,9 +16,12 @@ StorageManager .left.|> Storage
 StorageManager o-right-> UserPrefsStorage
 StorageManager o--> AddressBookStorage
 
+Storage -|> UserPrefsStorage
+Storage -|> AddressBookStorage
```
Comment (OhJunMing) [inline docs/diagrams/StorageClassDiagram.puml:20]: https://github.com/nus-cs2103-AY2021S2/forum/issues/224

Storage Interface only include 5 out of 8 methods from UserPrefsStorage, AddressBookStorage

File: docs/diagrams/ModelClassDiagram.puml [hunk 4]
```diff
@@ -5,12 +5,11 @@ skinparam arrowColor MODEL_COLOR
 skinparam classBackgroundColor MODEL_COLOR
 
 Package Model <<Rectangle>>{
-Interface ReadOnlyAddressBook <<Interface>>
-Interface Model <<Interface>>
-Interface ObservableList <<Interface>>
+Class "<<Interface>>\nReadOnlyAddressBook"
+Class "<<Interface>>\nModel"
+Class "<<Interface>>\nObservableList"
+
```
Comment (OhJunMing) [inline docs/diagrams/ModelClassDiagram.puml:11]: Interface class should be in non-italics
https://github.com/nus-cs2103-AY2021S2/forum/issues/276

**Review by nighoggDatatype — APPROVED** [review]
LGTM

**Pull request merged**